analysis/import/SCMLogParserV2/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparserv2/parser/LogLineParser.kt

```diff
@@ -0,0 +1,57 @@
+package de.maibornwolff.codecharta.importer.scmlogparserv2.parser
+
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.Commit
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.Modification
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.metrics.MetricsFactory
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.time.OffsetDateTime
+import java.util.stream.Stream
+
+/**
+ * Parses log lines and generates VersionControlledFiles from them using specific parserStrategy and metricsFactory
+ */
+class LogLineParser(
+    private val parserStrategy: LogParserStrategy,
+    private val metricsFactory: MetricsFactory,
+    private val silent: Boolean = false
+) {
+
+    private var numberOfCommitsParsed = 0
+
+    fun parse(logLines: Stream<String>): VersionControlledFilesList {
+        return logLines.collect(parserStrategy.createLogLineCollector())
+            .map { this.parseCommit(it) }.filter { !it.isEmpty }
```

Comment:
```suggestion
            .mapNotNull {
                val commit = this.parseCommit(it)
                if (!commit.isEmpty) {
                    commit
                }
            }
```

Reply:
Sadly not possible, as we can not call mapNotNull on streams